NDTensors/src/lib/Sectors/src/category_product.jl

```diff
@@ -68,44 +148,42 @@ function CategoryProduct(pairs::Pair...)
   return CategoryProduct(NamedTuple{keys}(vals))
 end
 
-function categories_fusion_rule(A::NamedTuple, B::NamedTuple)
-  qs = [A]
-  for (la, lb) in zip(pairs(intersect_keys(A, B)), pairs(intersect_keys(B, A)))
-    @assert la[1] == lb[1]
-    fused_vals = ⊗(la[2], lb[2])
-    qs = [union_keys((; la[1] => v), q) for v in fused_vals for q in qs]
-  end
-  # Include sectors of B not in A
-  qs = [union_keys(q, B) for q in qs]
-  return qs
-end
-
 function categories_equal(A::NamedTuple, B::NamedTuple)
```

Comment:
```suggestion
function categories_isequal(A::NamedTuple, B::NamedTuple)
```
to match the naming convention of Base Julia.

Reply:
Also this implementation seems quite complicated and seems like logic that should be handled in another way.

For example, we could use a code pattern like this:
```julia
function categories_insert_unspecified(a, b)
  # Insert trivial versions of unspecified categories into collections of categories `a` and `b`.
  # Can define versions for `NamedTuple` and `Tuple`.
  return a_specified, b_specified
end

function categories_isequal(a, b)
  return specified_categories_isequal(categories_insert_unspecified(a, b)...)
end

function specified_categories_isequal(a, b)
  return a == b
end
```
That pattern of using `categories_insert_unspecified` as a pre-processing step to call an inner function `specified_categories_...` could be a pattern used elsewhere as well, maybe it can help simplify some of the other code logic since handling the logic around category products having some unspecified categories seems to be a source of complexity in other parts of the code.

Reply:
see comment above for `isless`. I think we should keep the current `categories_isequal(o1::Tuple, o2::Tuple) = (o1 == o2)`, so the question is whether we accept a different convention for `==` with `NamedTuple`.